server/src/main/java/io/whitefox/core/services/DeltaSharedTable.java

```diff
@@ -77,6 +79,14 @@ public Optional<Long> getTableVersion(Optional<String> startingTimestamp) {
     return getSnapshot(startingTimestamp).map(Snapshot::getVersion);
   }
 
+  public boolean filterFilesBasedOnPredicates(List<String> predicates, AddFile f) {
```

Comment:
This is still not used, right?

Reply:
nope, I started on it, and figured it would be better to first make the predicates